src/nsrt_learning/side_predicate_learning.py

```diff
@@ -96,9 +96,27 @@ def _recompute_datastores_from_segments(
                             if not atoms.issubset(segment.final_atoms):
                                 continue
                         elif semantics == "add_effects":
+                            # Assuming we only have the ground op's add effects
+                            # filled in, we want all transitions in which the
+                            # demonstrator might have reasonably executed this
+                            # operator. In these transitions (1) the
+                            # preconditions of the ground op must hold,
+                            # (2) the add effects of the segment intersected
+                            # with the ground op's add effects must not be
+                            # empty [because otherwise, there's no point
+                            # calling this operator] and (3) all the ground
+                            # op's add effects must hold in the final atoms
+                            # [since these add effects are really just 'set'
+                            # effects].
+
                             if not ground_op.add_effects.issubset(
-                                    segment.add_effects):
+                                    segment.final_atoms):
+                                continue
+
+                            if not len(segment.add_effects
```

Comment:
change to `if len(segment.add_effects & ground_op.add_effects) == 0`?

Reply:
sure - either way works; I was just keeping with the `not` trend used in the previous statements lol

Reply:
you can do `if not (segment.add_effects & ground_op.add_effects)`